### DISCUSSION TITLE
Impact of change tracking

### DISCUSSION THREAD

**Discussion by SbiCA** [ravendb/ravendb#17954] [Q&A] [answered]
I was wondering that the impact of change tracking in a session might be https://ravendb.net/docs/article-page/6.0/csharp/client-api/session/configuration/how-to-disable-tracking 🤔  

did anyone measure this so far an see a significant change I'm wondering if there is a similar pattern to EF `AsNonTracking` that I wasn't aware of so far. 

It seems that in most cases where we optimised our code it's even bad since it introduce additional roundtrips due to disabled change tracker there fore (Include, Loads) don't seem to know about previously fetched things.

**Comment by ayende** [comment, marked as answer]
There is support for `NoTracking` if you want to, yes.
The benefit of not having that is reduced memory costs, but the downside is additional roundtrips and needing to maintain your own state.
In general, I would recommend not touching this

**Reply by SbiCA** [reply]
Yes, I agree I noticed the additional roundtrips as well since as part of our test we use `MaxNumberOfRequestsPerSession` to make sure we don't add more than necessary.

Do you already run benchmark this part of the client? 

I'm wondering because I recently stumbled across a few traces in our system where the time spent after being done with queries was pretty high. note I'm not saying it's the change tracking per se, I'm more interested in getting a general understanding and if it would be worth the time investment to setup benchmarks or some profiling sessions on our end.

**Reply by garayx** [reply]
Hi, 

unfortunately, we don't have any public benchmark on using session with `NoTracking` enabled, but I used session with `NoTracking` enabled, as part of running queries (internal) benchmark, and there was clear improvement in memory & cpu usage.
